### DISCUSSION TITLE
Is there an easy way (without changing the pion source code) to add info on a frame's packet

### DISCUSSION THREAD

**Discussion by amotzte** [pion/webrtc#1841] [Q&A] [answered]
https://github.com/pion/webrtc/blob/7948437b0b4633dd0ef30f4d7dbf1132ef23b716/track_local_static.go#L269-L274

```patch
	writeErrs := []error{}
+	packets[0].SetExtension(123,[]byte("My header extention")) // Add rtp header extension on first frame's packet
	for _, p := range packets {
		if err := s.rtpTrack.WriteRTP(p); err != nil {
			writeErrs = append(writeErrs, err)
		}
	}
```

And if there is no easy way, will you accept PR to add this ability ?

Thanks

**Comment by at-wat** [comment, marked as answer]
Hi @amotzte !
`TrackLocal` is [an interface](https://pkg.go.dev/github.com/pion/webrtc/v3#TrackLocal), so you can implement a custom `TrackLocal` (in this case, `webrtc.TrackLocalStaticSample` + your patch) in your package and pass it to `PeerConnection.AddTrack()` or `PeerConnection.AddTransceiverFromTrack()`.

**Comment by amotzte** [comment]
Hi @at-wat  
Thanks for the quick replay, will try it !
